core/modules/replica/tools/qserv-replica-job-chunks.cc

```diff
@@ -0,0 +1,384 @@
+
+        for (auto const& worker: provider->config()->workers()) {
+
+            std::string const numReplicasStr =
+                replicaData.workers.at(worker) ?
+                    std::to_string(worker2numChunks[worker]) :
+                    "*";
+
+            std::string const numQservReplicasStr =
+                pullQservReplicas and qservReplicaData.workers.at(worker) ?
+                    std::to_string(qservWorker2numChunks[worker]) :
+                    "*";
+
+            OUT << " " << std::setw(3)  << worker2idx[worker] << " |"
+                << " " << std::setw(40) << worker << " |"
+                << " " << std::setw(6)  << numReplicasStr << " |"
+                << " " << std::setw(6)  << numQservReplicasStr
+                << "\n";
+        }
+        OUT << "-----+------------------------------------------+--------+--------\n"
+            << std::endl;
+
+        OUT << "REPLICAS:\n"
+            << "  LEGEND:\n"
+            << "    for the desired minimal replication 'L'evel\n"
+            << "    for numbers of chunk replicas from different sources: 'R'eplication, 'Q'serv\n"
+            << "-------------+----------------------+-----+------+-----+------+------------------------------------------\n"
+            << "       chunk |             database |   R |  R-L |   Q |  Q-R | replicas at workers\n"
+            << "-------------+----------------------+-----+------+-----+------+------------------------------------------\n";
+
+        size_t const replicationLevel = provider->config()->replicationLevel(databaseFamily);
+
+        for (auto chunk: replicaData.chunks.chunkNumbers()) {
+            auto chunkMap = replicaData.chunks.chunk(chunk);
+
+            for (auto database: chunkMap.databaseNames()) {
+                auto databaseMap = chunkMap.database(database);
+
+                size_t      const  numReplicas        = databaseMap.size();
+                std::string const  numReplicasStr     = numReplicas ? std::to_string(numReplicas) : "";
+                long long   const  numReplicasDiff    = numReplicas - replicationLevel;
+                std::string const  numReplicasDiffStr = numReplicasDiff ? std::to_string(numReplicasDiff) : "";
+
+                OUT << " "   << std::setw(11) << chunk
+                    << " | " << std::setw(20) << database
+                    << " | " << std::setw(3)  << numReplicasStr
+                    << " | " << std::setw(4)  << numReplicasDiffStr;
+
+                if (pullQservReplicas) {
+                    size_t const numQservReplicas =
+                        qservReplicaData.useCount.chunkExists(chunk) and
+                        qservReplicaData.useCount.atChunk(chunk).databaseExists(database) ?
+                            qservReplicaData.useCount.atChunk(chunk).atDatabase(database).size() :
+                            0;
+
+                    std::string const numQservReplicasStr     = numQservReplicas ? std::to_string(numQservReplicas) : "";
```

Comment:
so if there's more than 0 it prints the string version of the number, otherwise it prints nothing (empty string)? I don't see the whole use case, but it seems odd.

Reply:
This is true. The empty string is printed in the corresponding table cell instead of 0. This is meant to produce a less noisy and easier to interpret output. Before ending up with the variant I had experimented with various options (including printing 0).